src/cryptsetup/cryptsetup.c

```diff
@@ -235,9 +235,25 @@ static int parse_one_option(const char *option) {
                 arg_readonly = true;
         else if (streq(option, "verify"))
                 arg_verify = true;
-        else if (streq(option, "silent"))
-                arg_silent = true;
-        else if (STR_IN_SET(option, "allow-discards", "discard"))
+        else if ((val = startswith(option, "password-echo="))) {
+                if (arg_ask_password_flags) {
+                        log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "Duplicate password-echo= option \"%s\", ignoring.", val);
+                        return 0;
+                }
```

Comment:
we usually don't check for duplicate parameters, but systematically override prior settings. This is useful if people extend existing options with new ones in naive ways.

hence, plesae drop this check here, and instead always set/unset *both* flags as appropriate.